File: veloce-backend/scripts/seed_users.py
"""Create default admin and reviewer if missing. Run: PYTHONPATH=. python scripts/seed_users.py"""

import asyncio
import os
from pathlib import Path

from dotenv import load_dotenv

load_dotenv(Path(__file__).resolve().parent.parent / ".env")

from app.db import init_db  # noqa: E402
from app.models.user import User  # noqa: E402
from app.passwords import hash_password  # noqa: E402


async def main() -> None:
    await init_db()

    admin_email = os.getenv("SEED_ADMIN_EMAIL", "admin@veloce.local").lower()
    admin_pass = os.getenv("SEED_ADMIN_PASSWORD", "admin123")
    admin_name = os.getenv("SEED_ADMIN_NAME", "Admin User")

    rev_email = os.getenv("SEED_REVIEWER_EMAIL", "reviewer@veloce.local").lower()
    rev_pass = os.getenv("SEED_REVIEWER_PASSWORD", "reviewer123")
    rev_name = os.getenv("SEED_REVIEWER_NAME", "Reviewer User")

    if not await User.find_one(User.email == admin_email):
        await User(
            email=admin_email,
            password_hash=hash_password(admin_pass),
            name=admin_name,
            role="admin",
        ).insert()
        print(f"Created admin: {admin_email}")
    else:
        print(f"Admin exists: {admin_email}")

    if not await User.find_one(User.email == rev_email):
        await User(
            email=rev_email,
            password_hash=hash_password(rev_pass),
            name=rev_name,
            role="reviewer",
        ).insert()
        print(f"Created reviewer: {rev_email}")
    else:
        print(f"Reviewer exists: {rev_email}")


if __name__ == "__main__":
    asyncio.run(main())

File: veloce-frontend/scripts/seed-users.ts
/**
 * Create default admin and reviewer if missing.
 * Run from veloce-frontend: npm run seed-users
 * Loads .env.local then .env (same vars as Vercel / former Python backend).
 */
import { config } from "dotenv";
import { MongoClient } from "mongodb";
import { resolve } from "path";

import { COLLECTIONS } from "../src/lib/server/collections";
import { hashPassword } from "../src/lib/server/passwords";

config({ path: resolve(process.cwd(), ".env.local") });
config({ path: resolve(process.cwd(), ".env") });

async function main() {
  const uri = process.env.MONGODB_URI;
  if (!uri) {
    console.error("Missing MONGODB_URI");
    process.exit(1);
  }
  const dbName = process.env.MONGODB_DB || "veloce";

  const adminEmail = (
    process.env.SEED_ADMIN_EMAIL || "admin@veloce.local"
  ).toLowerCase();
  const adminPass = process.env.SEED_ADMIN_PASSWORD || "admin123";
  const adminName = process.env.SEED_ADMIN_NAME || "Admin User";

  const revEmail = (
    process.env.SEED_REVIEWER_EMAIL || "reviewer@veloce.local"
  ).toLowerCase();
  const revPass = process.env.SEED_REVIEWER_PASSWORD || "reviewer123";
  const revName = process.env.SEED_REVIEWER_NAME || "Reviewer User";

  const client = new MongoClient(uri);
  await client.connect();
  const db = client.db(dbName);
  const users = db.collection(COLLECTIONS.users);

  if (!(await users.findOne({ email: adminEmail }))) {
    await users.insertOne({
      email: adminEmail,
      password_hash: hashPassword(adminPass),
      name: adminName,
      role: "admin",
    });
    console.log(`Created admin: ${adminEmail}`);
  } else {
    console.log(`Admin exists: ${adminEmail}`);
  }

  if (!(await users.findOne({ email: revEmail }))) {
    await users.insertOne({
      email: revEmail,
      password_hash: hashPassword(revPass),
      name: revName,
      role: "reviewer",
    });
    console.log(`Created reviewer: ${revEmail}`);
  } else {
    console.log(`Reviewer exists: ${revEmail}`);
  }

  await client.close();
}

main().catch((e) => {
  console.error(e);
  process.exit(1);
});
